### PR TITLE
[WIP] Add `picongpu --version` Flag

### PR DESCRIPTION
**Pull request by ax3l**
This pull request will add a new option

``` bash
  picongpu --version
```

or short `-v` that creates a short and pretty output of the dependencies that PIConGPU was compiled against.

**To do:**
- [ ] make more generic, so it can also be used for output files: https://github.com/openPMD/openPMD-standard/issues/137
- [ ] call should not return (not as `-h`) so it can be called in a single `mpiexec`
- [ ] it might be cool to output that before each rank from rank zero
- [ ] move output into own file (might avoid change to `.cu` which was necessary for `CUDA_VERSION` output)
- [ ] add `IceT` version
- [ ] add PIConGPU and libPMacc log levels
- [ ] compare to CUDA RT & output CUDA driver
- [ ] add `__TIME__` and `__DATE__` and author of compile process
- [ ] hostname and username on the current machine
- [ ] if compiled via `configure`, we could add if a `-t` case from `cmakeFlags` was used or even better, we store the `cmake_flags` from `configure` (#1100)

**Environment Changes:**
Increases the _minimum CMake version_ to **2.8.10+** to support [CMAKE_CXX_COMPILER_[ID|VERSION]](http://www.kitware.com/blog/home/post/408) on all platforms.

**Example:**

``` bash
$ ./build_picongpu/picongpu -v
PIConGPU:     0.1.2
  Build-Type: Debug

Third party:
  OS:         Linux-3.2.64 on x86_64
  CMAKE:      3.0.1 (Unix Makefiles)
  CUDA:       6050
  CXX:        GNU (4.6.2)
  boost:      105400
  MPI:
    standard: 2.1
    flavor:   OpenMPI (1.6.3)
  mallocMC:   2.1.0
  Splash:     1.2.4 (Format 2.1)
  ADIOS:      1.8.0
```

### CODE REVIEW

File: src/picongpu/CMakeLists.txt [hunk 1]
```diff
@@ -359,10 +359,29 @@ if(ADIOS_FOUND)
 
     include_directories(SYSTEM ${ADIOS_INCLUDE_DIRS})
     set(LIBS ${LIBS} ${ADIOS_LIBRARIES})
+    add_definitions(-DADIOS_VERSION="${ADIOS_VERSION}")
```
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: Only necessary for ADIOS < 1.9.0, see https://github.com/ornladios/ADIOS/commit/61cce400018a125933b85ae1b22555ccb2954d80

**Comment by PrometheusPi** [comment]
great idea :+1:

**Comment by bussmann** [comment]
Much awesome!

**Comment by n01r** [comment]
I assume you need to be on a machine with the right MPI module loaded, right? I'm referring to the same issue that exists with `./picongpu -h`.Can you only check what the executable has been compiled against if you have the correct libraries loaded?
That seems to be taking a bit of value from your new feature, if that is the case.
I would prefer to be able to do `./picongpu -h` and `./picongpu -v` on **any machine, regardless of which modules** I loaded. That would also make e.g. remote support much easier.

**Comment by ax3l** [comment]
that is, unfortunately, a problem that can barely be solved in the same binary. on other machines it is even harder, because there you need to cross-compile to a totally different hardware.

the only solution to that is to compile a second small binary with only that information, but the original idea is simply to output the version when the run starts and this is will already with the simple approach.

**Comment by n01r** [comment]
I see, too bad. Still, great work, though!

**Comment by ax3l** [comment]
yes, but still, the idea is to get it documented in `stdout` and that will work fine and reliable.

**Comment by n01r** [comment]
Yeah, sure ^^ I was just thinking:
*What if I encounter a wild picongpu executable while walking through high grass? It suddenly attacks me and because I am in unknown territory I do not know which libraries I need to execute it*

**Comment by ax3l** [comment]
Hm, I see. I guess that depends on a lot of things but mainly on *your trust to Prof. Oak in providing you with a save binary for walking through high grass. Nevertheless, you should not leave pallet town without a compiler in the first place, it is just too dangerous out there*.

**Pull request closed without merging**